### PR TITLE
ENG-1149

### PR DESCRIPTION
**Pull request by IvanKaraslavov**
Adds release dates on content level so all types of content can have them.

### CODE REVIEW

File: src/main/java/org/atlasapi/input/ContentModelTransformer.java [hunk 1]
```diff
@@ -270,4 +274,20 @@ private CrewMember transformPerson(Person person, Publisher publisher) {
...
+            ReleaseDate complexReleaseDate = new ReleaseDate(convertToLocalDate(releaseDate.getDate()),
+                    Countries.fromCode(releaseDate.getCountry()),
+                    ReleaseDate.ReleaseType.valueOf(releaseDate.getType().toUpperCase()));
+            builder.add(complexReleaseDate);
+        }
+        return builder.build();
+    }
+
+    private LocalDate convertToLocalDate(Date date) {
+        return new LocalDate(date);
```
Comment (tristanbeardmore-gray) [inline src/main/java/org/atlasapi/input/ContentModelTransformer.java:291]: Why did you remove the null check?

File: src/main/java/org/atlasapi/query/content/merge/ContentMerger.java [hunk 2]
```diff
@@ -257,4 +262,13 @@ private Content merge(
     private <T> List<T> merge(List<T> existing, List<T> posted) {
         return ImmutableSet.copyOf(Iterables.concat(posted, existing)).asList();
     }
+
+    public Item mergeReleaseDates(Item existing, Item update, boolean merge) {
```
Comment (tristanbeardmore-gray) [inline src/main/java/org/atlasapi/query/content/merge/ContentMerger.java:266]: It doesn't look like this is used any more.

File: src/main/java/org/atlasapi/input/ContentModelTransformer.java (hunk 1 above)
Comment (tristanbeardmore-gray) [reply src/main/java/org/atlasapi/input/ContentModelTransformer.java:291]: The complex `ReleaseDate` does enforce the date to be non-null but you're better off letting that handle raising the exception instead of inadvertently NPEing here.

Comment (pawel-mtbcst) [reply src/main/java/org/atlasapi/input/ContentModelTransformer.java:291]: @tristanbeardmore-gray if we're sure it won't be null, why write the code that is not needed?
We could add @nonNull annotation on the param. Also, this method can be static, or even can be moved to some Utility class.

File: src/main/java/org/atlasapi/query/content/merge/ContentMerger.java [hunk 3]
```diff
@@ -257,4 +262,13 @@ private Content merge(
     private <T> List<T> merge(List<T> existing, List<T> posted) {
         return ImmutableSet.copyOf(Iterables.concat(posted, existing)).asList();
```
Comment (pawel-mtbcst) [inline src/main/java/org/atlasapi/query/content/merge/ContentMerger.java:263]: Alternative using JDK methods, that does not create set just to remove duplicates:
`concat(existing.stream(), posted.stream()).distinct().collect(toList())`

File: src/main/java/org/atlasapi/input/ContentModelTransformer.java (hunk 1 above)
Comment (tristanbeardmore-gray) [reply src/main/java/org/atlasapi/input/ContentModelTransformer.java:294]: This class essentially converts between two very similar but different data models (from the 'simple' model to the  'complex' model). It looks like the date field is nullable for the simple model but not the complex. I don't consider it this class's responsibility to care for this difference, and the validation can be left within the simple or complex model class itself.

That said this is a rather minor point, especially because the exception that would be thrown otherwise from the complex model class doesn't really provide any additional information about the exception (unless the IllegalArgumentException that gets thrown is caught and converted into a meaningful HTTP response whilst the NPE possibly doesn't - but I don't believe this is the case).
